### PR TITLE
feat(docs): clean up app transfer docs

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

- follow up on "further work" from https://github.com/expo/expo/pull/18926

# How

Utilizing @brentvatne's feedback I deleted most of the extraneous info, then I realized we should just link to the actual guide in our docs.

### CODE REVIEW

File: docs/pages/distribution/app-transfers.md [hunk 1]
```diff
@@ -1,13 +1,19 @@
...
+import { BoxLink } from '~/ui/components/BoxLink';
 
-One of the [criteria for an app to be eligible to transfer](https://help.apple.com/app-store-connect/#/devaf27784ff) is that `"No version of the app can use a Passbook entitlement."`. Until December 11, 2018, apps built with Expo SDK 30 or 31 included the `Payments` module, which necessitates linking a file called `PassKit.framework`. This file includes the native APIs for interacting with the native iOS payments functionality. All SDK 30 and 31 apps built before this date included this file -- this is because we always include the native code for all modules in every build, even modules that you don't import in your JS. (Among other reasons, this makes it possible for you to use different modules in updates without having to build a new binary.)
+Universal apps using EAS have two parts:
```
Comment (brentvatne) [inline docs/pages/distribution/app-transfers.md:8]:
```suggestion
There are two different representations of your app to consider when handing over ownership to another entity: the app as it exists on Expo Application Services (in order to create builds with EAS Build, send updates with EAS Update, etc.) and the app records on the Apple App Store / Google Play Store (in order to distribute the app to end-users). The following guides explain how to handle app transfers in each of these cases.
```

File: docs/pages/distribution/app-transfers.md [hunk 2]
```diff
@@ -1,13 +1,19 @@
...
 
-While no apps built with the classic build service have ever had any Passbook entitlements enabled, we've received a number of reports indicating that simply including `PassKit.framework` in a project is enough to make an app permanently ineligible to transfer. We removed the `Payments` module from all iOS builds beginning December 12, 2018. However, if you built an SDK 30 or 31 iOS app with `expo build:ios` before this date, and uploaded the resulting .ipa file to App Store Connect, your app will likely be ineligible to transfer using the App Transfer process. If you attempt to transfer your app but receive a rejection due to "use of the PassKit Integration API", then your app is affected by this issue.
+1. JavaScript files and assets which are hosted with [EAS Update](/eas-update/introduction). These static files are used for over-the-air updates.
+2. [Native runtime](/workflow/glossary-of-terms#native-runtime) hosted in the respective app stores. Used for rendering the static hosted assets.
```
Comment (brentvatne) [inline docs/pages/distribution/app-transfers.md:11]:
```suggestion
```

File: docs/pages/distribution/app-transfers.md [hunk 3]
```diff
@@ -1,13 +1,19 @@
...
 
-For affected apps, the only way to transfer an app to a different account is to resubmit the app under the new Apple Developer account with a new `bundleIdentifier` (binary reassignment). The transferred app will effectively be a new and separate app on the App Store and on users' devices. We suggest that once this process is completed, you submit a final update to the old app which notifies users about the new app listing and directs them to download it.
+Learn more in the official guides for each:
```
Comment (brentvatne) [inline docs/pages/distribution/app-transfers.md:13]:
```suggestion
```

**Review by brentvatne — CHANGES_REQUESTED** [review]

File: docs/pages/distribution/app-transfers.md [hunk 4]
```diff
@@ -1,13 +1,16 @@
...
 
-One of the [criteria for an app to be eligible to transfer](https://help.apple.com/app-store-connect/#/devaf27784ff) is that `"No version of the app can use a Passbook entitlement."`. Until December 11, 2018, apps built with Expo SDK 30 or 31 included the `Payments` module, which necessitates linking a file called `PassKit.framework`. This file includes the native APIs for interacting with the native iOS payments functionality. All SDK 30 and 31 apps built before this date included this file -- this is because we always include the native code for all modules in every build, even modules that you don't import in your JS. (Among other reasons, this makes it possible for you to use different modules in updates without having to build a new binary.)
+There are two different representations of your app to consider when handing over ownership to another entity: the app as it exists on Expo Application Services (in order to create builds with EAS Build, send updates with EAS Update, etc.) and the app records on the Apple App Store / Google Play Store (in order to distribute the app to end-users). The following guides explain how to handle app transfers in each of these cases.
```
Comment (amandeepmittal) [inline docs/pages/distribution/app-transfers.md:8]:
```suggestion
There are two different representations of your app to consider when handing over ownership to another entity: the app as it exists on Expo Application Services (to create builds with EAS Build, send updates with EAS Update, etc.) and the app records on the Apple App Store/Google Play Store (to distribute the app to end-users). The following guides explain how to handle app transfers in each case.
```

As per our style guide, let's [avoid adding spaces before and after `/`](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#do-not-add-a-space-between-the-words-and-).

**Review by amandeepmittal — APPROVED** [review]
Left a small change to follow our styleguide. Rest looks good to me 👍

**Pull request merged**